process/smartContract/common.go

```diff
@@ -0,0 +1,38 @@
+package smartContract
+
+import (
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/process"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+)
+
+func findVMByTransaction(container process.VirtualMachinesContainer, tx data.TransactionHandler) (vmcommon.VMExecutionHandler, error) {
+	scAddress := tx.GetRcvAddr()
+	return findVMByScAddress(container, scAddress)
+}
+
+func findVMByScAddress(container process.VirtualMachinesContainer, scAddress []byte) (vmcommon.VMExecutionHandler, error) {
+	vmType, err := parseVMTypeFromContractAddress(scAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	vm, err := container.Get(vmType)
+	if err != nil {
+		return nil, err
+	}
+
+	return vm, nil
+}
+
+func parseVMTypeFromContractAddress(contractAddress []byte) ([]byte, error) {
+	// TODO: Why not check against AddressLength (32)?
+	if len(contractAddress) < core.NumInitCharactersForScAddress {
```

Comment:
why did you separate this against 3 separate function ? plus it was ok in the core/address.go the function as it was.

Reply:
The refactoring was to bring the `vmType` logic where used, and then to remove a tiny bit of duplication from `scQueryService` and `process`.

Currently, 
 - `findVMByTransaction` used by `process.go`
 - `findVMByScAddress` used by `scQueryService` (where there is no actual transaction)
 - `parseVMTypeFromContractAddress` was extracted separately to encapsulate the parsing of `vmType`.

Is it ok like this?

Reply:
ok. it makes sense :)